Source/monster.cpp

```diff
@@ -6010,29 +5976,26 @@ void __fastcall MAI_Warlord(int i)
 
 void __cdecl DeleteMonsterList()
 {
-	int *v0;       // eax
-	signed int v1; // ecx
-
-	v0 = &monster[0]._my;
-	do {
-		if (v0[18]) {
-			*(v0 - 1) = 1;
-			*v0 = 0;
-			v0[1] = 0;
-			v0[2] = 0;
-			v0[3] = 0;
-			v0[4] = 0;
-			v0[18] = 0;
-		}
-		v0 += 57;
-	} while ((signed int)v0 < (signed int)&monster[4]._my);
-	v1 = 4;
-	while (v1 < nummonsters) {
-		if (monster[monstactive[v1]]._mDelFlag) {
-			DeleteMonster(v1);
-			v1 = 0;
+	int i;
+	for (i = 0; i < 4; i++) {
+		if (monster[i]._mDelFlag) {
+			monster[i]._mx = 1;
+			monster[i]._my = 0;
+			monster[i]._mfutx = 0;
+			monster[i]._mfuty = 0;
+			monster[i]._moldx = 0;
+			monster[i]._moldy = 0;
+			monster[i]._mDelFlag = FALSE;
+		}
+	}
+
+	i = 4;
+	while (i < nummonsters) {
+		if (monster[monstactive[i]]._mDelFlag) {
+			DeleteMonster(i);
+			i = 0;

```

Comment:
Is this a bug? The first 4 monsters seams reserved for quest monsters (wasn't the case for the beta), but this reset going all the way back to 0.

Reply:
It could possibly be a bug, in which case `i = 4;` seems to be the correct version?